browser/components/smartwindow/content/insights.mjs

```diff
@@ -983,6 +1034,29 @@ export function createInsightsOverlay(
     window.dispatchEvent(new CustomEvent("insights-updated"));
   };
 
+  const handleGenerateInsightsWithLLM = async (event) => {
+    // Get the input element from the event target's parent
+    const input = event.target.parentElement.querySelector("#llm-insights-input");
+    const text = input?.value?.trim();
+
+    if (!text) {
+      console.warn("No text provided for LLM insights generation");
+      return;
+    }
+
+    try {
+      await generateInsightsFromCustomText(text);
+      // Clear the input after successful generation
+      if (input) {
+        input.value = "";
+      }
+      // Force re-render by triggering a state change
+      window.dispatchEvent(new CustomEvent("insights-updated"));
```

Comment:
i moved this outside so errors show up as well https://github.com/Firefox-AI/firefox-prototypes/commit/fa65a533d111ab0c2f77483cc5ba536dafe29929